From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Tue, 30 Apr 2024 13:17:46 -0500
Subject: [PATCH] solus: Suppress -dirty from version string

This prevents `-dirty` from being appended to the version string, which just looks bad and also isn't really relevant since we build from git and our local patches are build fixes.
---
 CMake/ScmRevGen.cmake               | 2 +-
 Externals/mGBA/make_version.c.js    | 4 ++--
 Source/Core/Common/make_scmrev.h.js | 2 +-
 3 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/CMake/ScmRevGen.cmake b/CMake/ScmRevGen.cmake
index 5ec3041134..1cc6687d89 100644
--- a/CMake/ScmRevGen.cmake
+++ b/CMake/ScmRevGen.cmake
@@ -7,7 +7,7 @@ if(GIT_FOUND)
       OUTPUT_VARIABLE DOLPHIN_WC_REVISION
       OUTPUT_STRIP_TRAILING_WHITESPACE)
   # defines DOLPHIN_WC_DESCRIBE
-  execute_process(WORKING_DIRECTORY ${PROJECT_SOURCE_DIR} COMMAND ${GIT_EXECUTABLE} describe --always --long --dirty
+  execute_process(WORKING_DIRECTORY ${PROJECT_SOURCE_DIR} COMMAND ${GIT_EXECUTABLE} describe --always --long
       OUTPUT_VARIABLE DOLPHIN_WC_DESCRIBE
       OUTPUT_STRIP_TRAILING_WHITESPACE)
 
diff --git a/Externals/mGBA/make_version.c.js b/Externals/mGBA/make_version.c.js
index d52d9e9a05..ee41bd8a7f 100644
--- a/Externals/mGBA/make_version.c.js
+++ b/Externals/mGBA/make_version.c.js
@@ -3,8 +3,8 @@ var oFS				= new ActiveXObject("Scripting.FileSystemObject");
 
 wshShell.CurrentDirectory   += "\\mgba";
 var outfile                 = "../version.c";
-var cmd_commit              = " describe --always --abbrev=40 --dirty";
-var cmd_commit_short        = " describe --always --dirty";
+var cmd_commit              = " describe --always --abbrev=40";
+var cmd_commit_short        = " describe --always";
 var cmd_branch              = " symbolic-ref --short HEAD";
 var cmd_rev                 = " rev-list HEAD --count";
 var cmd_tag                 = " describe --tag --exact-match";
diff --git a/Source/Core/Common/make_scmrev.h.js b/Source/Core/Common/make_scmrev.h.js
index 298f4e0505..b320a59bd7 100644
--- a/Source/Core/Common/make_scmrev.h.js
+++ b/Source/Core/Common/make_scmrev.h.js
@@ -3,7 +3,7 @@ var oFS				= new ActiveXObject("Scripting.FileSystemObject");
 
 var outfile			= "./scmrev.h";
 var cmd_revision	= " rev-parse HEAD";
-var cmd_describe	= " describe --always --long --dirty";
+var cmd_describe	= " describe --always --long";
 var cmd_branch		= " rev-parse --abbrev-ref HEAD";
 var cmd_commits_ahead = " rev-list --count HEAD ^master";
 
